### PR TITLE
make it possible tu use :GpWhisper <LANGUAGE>

### PR DESCRIPTION
**Pull request by teto**
so this can be changed on the fly.

Ideally I would like to be able to pass it an existing file too but I wanted to do a minimum first contribution. Also I develop on top of https://github.com/Robitx/gp.nvim/pull/93 which doesn't help

### CODE REVIEW

**Comment by teto** [comment]
@Robitx you mentioned pending comments but I dont see any ? could you precise please

**Review by Robitx — CHANGES_REQUESTED** [review]

**Comment by Robitx** [comment]
@teto sorry, I've forgot the review in the pending state :man_facepalming:

**Comment by teto** [comment]
ha yeah the log commands interrupt the user, quite bad ^^'''
I wish we could have logging to a file though, it's just too helpful to diagnose issues.
What do you think of vendoring https://github.com/nvim-neorocks/rocks.nvim/blob/master/lua/rocks/log.lua (or something else but there doesn't seem to be a clear winner among logging library yet, plenary.nvim''s is bad) ? could be for another PR or this one.

File: lua/gp/init.lua [hunk 1]
```diff
@@ -2793,7 +2794,8 @@ M.Prompt = function(params, target, prompt, model, template, system_template, wh
 end
 
 ---@param callback function # callback function(text)
-M.Whisper = function(callback)
+M.Whisper = function(language, callback)
+	M.info("Whisper language set to "..language)
```
[the remark this answers is not in the document]
Comment (teto) [reply lua/gp/init.lua]: removed

File: lua/gp/init.lua [hunk 2]
```diff
@@ -248,6 +248,7 @@ _H.process = function(buf, cmd, args, callback, out_reader, err_reader)
 		M.remove_handle(pid)
 	end))
 
+	M.info("spawning command: "..cmd)
```
[the remark this answers is not in the document]
Comment (teto) [reply lua/gp/init.lua]: removed

File: lua/gp/init.lua [hunk 3]
```diff
@@ -982,7 +983,7 @@ M.prepare_commands = function()
 		end
 
 		M.cmd["Whisper" .. command] = function(params)
-			M.Whisper(function(text)
+			M.Whisper("en", function(text)
```
[the remark this answers is not in the document]
Comment (teto) [reply lua/gp/init.lua]: yes ! done

**Comment by teto** [comment]
rebased, addressed comments and updated the doc. Tested locally with French and English. Worked nicely.

> For logging I wouldn't drag another dependency just for this

I was not proposing to add a dependency (yet :) ) but just to copy/paste that file into our codebase.

**Comment by Robitx** [comment]
@teto btw. the `doc/gp.nvim.txt` is made automatically by a CI function, README is the source of truth from which it's made.

**Review by Robitx — APPROVED** [review]

**Pull request merged**